forge-gui/res/cardsfolder/upcoming/saruman_of_many_colors.txt

```diff
@@ -0,0 +1,16 @@
+Name:Saruman of Many Colors
+ManaCost:3 W U B
+Types:Legendary Creature Avatar Wizard
+PT:5/4
+K:Ward:Discard<1/Enchantment;Instant;Sorcery/enchantment, instant or sorcery card>
+T:Mode$ SpellCast | ValidCard$ Card.YouCtrl | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigMill | ActivatorThisTurnCast$ EQ2 | TriggerDescription$ Whenever you cast your second spell each turn, each opponent mills two cards. When one or more cards are milled this way, exile target enchantment, instant, or sorcery card with equal or lesser mana value than that spell from an opponent's graveyard. Copy the exiled card. You may cast the copy without paying its mana cost.
+SVar:TrigMill:DB$ Mill | Defined$ Opponent | NumCards$ 2 | RememberMilled$ True | SubAbility$ DBImmediateTrigger
+SVar:DBImmediateTrigger:DB$ ImmediateTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card | RememberObjects$ TriggeredCard | SubAbility$ DBCleanup | TriggerDescription$ When one or more cards are milled this way, exile target enchantment, instant, or sorcery card with equal or lesser mana value than that spell from an opponent's graveyard. Copy the exiled card. You may cast the copy without paying its mana cost.
+SVar:TrigExile:DB$ ChangeZone | Origin$ Graveyard | Destination$ Exile | TgtZone$ Graveyard | ValidTgts$ Instant.OppOwn+cmcLEX,Sorcery.OppOwn+cmcLEX  | TgtPrompt$ Select target instant or sorcery card from an opponent's graveyard | Imprint$ True | SubAbility$ DBPlay
+SVar:DBPlay:DB$ Play | Valid$ Card.IsImprinted | ValidZone$ Exile | Controller$ You | CopyCard$ True | WithoutManaCost$ True | ValidSA$ Spell | Optional$ True | Amount$ All | SubAbility$ DBCleanupBis
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:DBCleanupBis:DB$ Cleanup | ClearImprinted$ True
+SVar:X:Remembered$CardManaCost
```

Comment:
A trigger doesn't have it's own host that's why you need to use TriggerRemembered

Reply:
Thanks for all your help but sadly card is still not working 😞

Reply:
That's because you also randomly removed Execute$

Reply:
Oh god no wonder it didn't work 🤦

Reply:
Finally got it working, thanks again 
![image](https://github.com/Card-Forge/forge/assets/67333662/dc3ca730-c34b-46e3-9d3c-a6c447c57736)

Reply:
Still something wrong with it ?

Reply:
Chance is you're not handling LKI correctly yet

Check with X spell that you counter before resolving the trigger

Reply:
It did still allow me to use Sarumans ability. I guess that's not allowed ? Does it need RememberObjects$ TriggeredCardLKICopy ?

Reply:
Nothing in the ability forbids the game from using LKI to obtain the value
so yea, that change is probably the right direction